src/plan_ordered_append.c

```diff
@@ -106,6 +106,12 @@ ts_ordered_append_path_create(PlannerInfo *root, RelOptInfo *rel, Hypertable *ht
 	if (list_length(merge->subpaths) == 0)
 		return NULL;
 
+#if !(PG96 || PG10)
+	/* Turn off partitioning. Otherwise, `apply_scanjoin_target_to_paths` will blow away these paths
+	 * and will not call hooks to redo this */
+	rel->nparts = 0;
```

Comment:
Again, seems like this should be more of a top-level decision by the planner rather than delegated to individual plan nodes. A top-level decision would also avoid having to do this at multiple places in plan nodes.